rand_distr/src/rayleigh.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! The Rayleigh distribution.
+use rand::Rng;
+use crate::{Distribution, OpenClosed01};
+use crate::utils::Float;
+
+#[derive(Clone, Copy, Debug)]
+struct Rayleigh<N> {
+	sigma: N
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Error {
+    /// `sigma < 0` or `nan`.
+    SigmaTooSmall
+}
+
+impl<N:Float> Rayleigh<N>{
+	pub fn new(sigma:N) -> Result<Rayleigh<N>,Error>{
+		if !(sigma <= N::from(0.0)) {
+            return Err(Error::SigmaTooSmall);
+        }
+        Ok(Rayleigh{sigma:sigma})
+	}
+}
+
+impl<N: Float> Distribution<N> for Rayleigh<N>
+where OpenClosed01: Distribution<N>
+{
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> N {
+        let x: N = rng.sample(OpenClosed01);
+        let exp_power = -self.x.powi(2)/(2*self.sigma.powi(2));
+        x*exp_power.exp()/self.sigma.powi(2)
```

Comment:
I think this can be further simplified, see the [equivalent implementation of the Weibull distribution](https://github.com/rust-random/rand/blob/19b5a3e85cae272602abb51b5cae511980fff47c/rand_distr/src/weibull.rs#L60).

Reply:
`        let sigma_square = self.sigma.powf(N::from(2.0));
        let exp_power = -x.powf(N::from(2.0))/(2*sigma_square);
        (x*exp_power.exp())/sigma_square;`
what about this..should I try to be more simplified than this.

Reply:
I Float trait doesn't have powi.. so you want me add it to the trait or just create function powi in rayliegh.rs ?

Reply:
> I think this can be further simplified, see the [equivalent implementation of the Weibull distribution](https://github.com/rust-random/rand/blob/19b5a3e85cae272602abb51b5cae511980fff47c/rand_distr/src/weibull.rs#L60).

No you can have a look I change to multiplication